src/main/kotlin/brs/db/sql/SqlIndirectIncomingStore.kt

```diff
@@ -33,30 +32,33 @@ internal class SqlIndirectIncomingStore(private val dp: DependencyProvider) : In
                 )
             }
 
-            private fun getQuery(ctx: DSLContext, indirectIncoming: IndirectIncoming): Query {
-                val record = IndirectIncomingRecord()
-                record.accountId = indirectIncoming.accountId
-                record.transactionId = indirectIncoming.transactionId
-                record.height = indirectIncoming.height
-                return ctx.upsert(record, INDIRECT_INCOMING.ACCOUNT_ID, INDIRECT_INCOMING.TRANSACTION_ID)
-            }
-
             override fun save(ctx: DSLContext, entity: IndirectIncoming) {
-                getQuery(ctx, entity).execute()
-            }
-
-            override fun save(ctx: DSLContext, entities: Array<IndirectIncoming>) {
-                val queries = mutableListOf<Query>()
-                for (indirectIncoming in entities) {
-                    queries.add(getQuery(ctx, indirectIncoming))
-                }
-                ctx.batch(queries).execute()
+                val record = IndirectIncomingRecord()
+                record.accountId = entity.accountId
+                record.transactionId = entity.transactionId
+                record.height = entity.height
+                ctx.upsert(record, INDIRECT_INCOMING.ACCOUNT_ID, INDIRECT_INCOMING.TRANSACTION_ID)
             }
         }
     }
 
     override fun addIndirectIncomings(indirectIncomings: Collection<IndirectIncoming>) {
-        dp.db.useDslContext { ctx -> indirectIncomingTable.save(ctx, indirectIncomings.toTypedArray()) }
+        dp.db.useDslContext { ctx ->
+            var insertQuery = ctx.insertInto(
+                INDIRECT_INCOMING,
+                INDIRECT_INCOMING.ACCOUNT_ID,
+                INDIRECT_INCOMING.TRANSACTION_ID,
+                INDIRECT_INCOMING.HEIGHT
+            )
+            indirectIncomings.toList().map { entity ->
```

Comment:
`toList().map` here can be replaced with `forEach` - the toList() is actually creating a new collection under the hood which is unnecessary and the map is too, which is also unnecessary

Reply:
Thx, I will read about kotlin :)

Reply:
looks like after new changes doesn't needed do it

Reply:
ok I understood you, I took these code from transaction batch, but there it was really needed